test/base/trainopts.js

```diff
@@ -74,102 +75,108 @@ describe('train() options', () => {
   });
 });
 
-
-describe('async train() options', () => {
-  it('train until error threshold reached', (done) => {
-    let net = new brain.NeuralNetwork();
-    let p1 = net
-      .trainAsync(data, { errorThresh: 0.2 })
-      .then (res => {
-        assert.ok(res.error < 0.2, `[res.error, ${res.error}] should have been less then 0.2`);
+describe('train() and trainAsync() use the same private methods', () => {
+  let trainingData = [{ input: [0, 0], output: [0] }];
+  let opts = { iterations:1 };
+  let net = new brain.NeuralNetwork();
+
+  function getSpy (net, func) {
```

Comment:
because of how simple it is, can you just call `sinon.spy` and `.restore()` directly?

Reply:
but what if I want to kill the spy?